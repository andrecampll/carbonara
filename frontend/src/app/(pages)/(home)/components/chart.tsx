'use client'

import { Button, Card, CardContent, Typography } from '@mui/material'
import { PieChart } from '@mui/x-charts'

import { useFootprint } from '@/hooks/use-footprint'

export function Chart() {
  const { footprintChartData, footprint, clearFootprint } = useFootprint()

  return footprintChartData.length > 0 ? (
    <Card>
      <CardContent>
        <Typography variant="h5">Your carbon footprint</Typography>

        <div className="mt-8 flex w-full items-center">
          <PieChart
            series={[
              {
                data: footprintChartData,
                highlightScope: { faded: 'global', highlighted: 'item' },
                faded: {
                  innerRadius: 30,
                  additionalRadius: -30,
                  color: 'gray',
                },
              },
            ]}
            width={600}
            height={300}
            colors={['#007EA7', '#1DD1A0', '#9F7E69', '#FFCAE9']}
          />
        </div>
      </CardContent>

      <CardContent>
        <Typography variant="h6">Carbon Footprint Breakdown</Typography>

        <ul className="mt-4">
          {footprintChartData.map((data) => (
            <li key={data.id} className="flex justify-between">
              <Typography fontWeight="600">{data.label}</Typography>

              <span className="flex items-center gap-2">
                <Typography fontSize={20}>{data.value.toFixed(2)}</Typography>

                <strong>{data.unit || 'kg'}</strong>
              </span>
            </li>
          ))}

          <li className="flex justify-between">
            <Typography fontWeight="600">Total Emissions</Typography>

            <span className="flex items-center gap-2">
              <Typography fontSize={20}>
                {footprint?.totalYearlyEmissions?.value.toFixed(2)}
              </Typography>

              <strong>{footprint?.totalYearlyEmissions?.unit || 'kg'}</strong>
            </span>
          </li>
        </ul>
      </CardContent>

      <CardContent>
        <Typography>
          Your carbon footprint is calculated based on the data you provided.
          The data is calculated using the latest carbon footprint calculation
          methods and is an estimate. The data is not 100% accurate and should
          be used as a reference only.
        </Typography>{' '}
        <br />
        <Typography>
          You can test different scenarios by changing the data you provided and
          see how it affects your carbon footprint. You can also see how
          different choices can help you reduce your carbon footprint.
        </Typography>
      </CardContent>

      <CardContent className="flex w-full justify-center">
        <Button onClick={clearFootprint} variant="contained" color="secondary">
          Try again with different values
        </Button>
      </CardContent>
    </Card>
  ) : null
}
