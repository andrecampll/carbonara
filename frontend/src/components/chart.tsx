'use client'

import { Card, CardContent, Typography } from '@mui/material'
import { PieChart } from '@mui/x-charts'

import { useFootprint } from '@/hooks/use-footprint'

export function Chart() {
  const { footprintChartData, footprint } = useFootprint()

  return footprintChartData.length > 0 ? (
    <Card>
      <CardContent>
        <Typography variant="h5">Your carbon footprint</Typography>

        <div className="mt-8 flex w-full items-center">
          <PieChart
            series={[
              {
                data: footprintChartData,
                highlightScope: { faded: 'global', highlighted: 'item' },
                faded: {
                  innerRadius: 30,
                  additionalRadius: -30,
                  color: 'gray',
                },
              },
            ]}
            width={600}
            height={300}
            colors={['#007EA7', '#1DD1A0', '#9F7E69', '#FFCAE9']}
          />
        </div>
      </CardContent>

      <CardContent>
        <ul className="mt-4">
          {footprintChartData.map((data) => (
            <li key={data.id} className="flex justify-between">
              <Typography fontWeight="600">{data.label}</Typography>

              <span className="flex items-center gap-2">
                <Typography fontSize={20}>{data.value.toFixed(2)}</Typography>

                <strong>{data.unit || 'kg'}</strong>
              </span>
            </li>
          ))}

          <li className="flex justify-between">
            <Typography fontWeight="600">Total Emissions</Typography>

            <span className="flex items-center gap-2">
              <Typography fontSize={20}>
                {footprint?.totalYearlyEmissions?.value.toFixed(2)}
              </Typography>

              <strong>{footprint?.totalYearlyEmissions?.unit || 'kg'}</strong>
            </span>
          </li>
        </ul>
      </CardContent>
    </Card>
  ) : null
}
