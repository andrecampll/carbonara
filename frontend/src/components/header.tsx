// import Image from 'next/image'
import Link from 'next/link'

export function Header() {
  return (
    <header className="bg-gradient-to-r from-primary-500 to-secondary-500 p-4 md:px-12">
      <div className="m-auto flex max-w-5xl items-center justify-between">
        <Link href="/">
          <div className="flex items-center gap-2">
            {/* <Image alt="logo" width={30} height={30} src="/logo.png" /> */}
            <span className="text-xl font-bold text-white">Carbonara</span>
          </div>
        </Link>

        <div className="flex items-center gap-4">
          <p className="text-white">John Doe</p>
          <div className="rounded-full bg-white p-1">
            <span className="font-bold text-black">JD</span>
          </div>
        </div>
      </div>
    </header>
  )
}
