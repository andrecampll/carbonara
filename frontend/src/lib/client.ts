import { ApolloClient, HttpLink, InMemoryCache } from '@apollo/client'
import { registerApolloClient } from '@apollo/experimental-nextjs-app-support/rsc'

import { env } from '@/env'

export const { getClient } = registerApolloClient(() => {
  return new ApolloClient({
    cache: new InMemoryCache(),
    link: new HttpLink({
      uri: env.NEXT_PUBLIC_API_BASE_URL,
    }),
  })
})
