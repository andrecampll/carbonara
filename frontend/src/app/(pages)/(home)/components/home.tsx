'use client'

import { Chart } from '@/components/chart'
import { FormWizard } from '@/components/wizard/form-wizard'
import { useFootprint } from '@/hooks/use-footprint'

export function Home() {
  const { footprint } = useFootprint()

  return (
    <div className="my-6 flex w-full flex-col gap-4 lg:flex-row">
      {!footprint ? <FormWizard /> : <Chart />}
    </div>
  )
}
