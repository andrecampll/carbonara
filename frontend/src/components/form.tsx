'use client'

import { useMutation } from '@apollo/client'
import { zodResolver } from '@hookform/resolvers/zod'
import { Card, CardContent, Stack, Typography } from '@mui/material'
import Button from '@mui/material/Button'
import { useCallback } from 'react'
import { useForm } from 'react-hook-form'
import { z } from 'zod'

import { Footprint } from '@/data/types/footprint'
import { CALCULATE_FOOTPRINT_MUTATION } from '@/graphql/mutations/calculateFootprint'
import { useFootprint } from '@/hooks/use-footprint'

import { Input } from './ui/input'
import { SearchInput } from './ui/search-input'

const footprintFormSchema = z.object({
  electricty: z.string().min(1, {
    message: 'Electricity usage is required',
  }),
  transportation: z.string().min(1, {
    message: 'Transportation gasoline usage is required',
  }),
  shortFlights: z.string().min(1, {
    message: 'Short flights is required',
  }),
  mediumFlights: z.string().min(1, {
    message: 'Medium flights is required',
  }),
  longFlights: z.string().min(1, {
    message: 'Long flights is required',
  }),
  dietaryChoice: z.string().min(1, {
    message: 'Dietary choice is required',
  }),
})

type FootprintFormData = z.infer<typeof footprintFormSchema>

type CalculateFootprintMutationResponse = {
  calculate: {
    footprint: Footprint
  }
}

export function Form() {
  const {
    register,
    handleSubmit,
    formState: { errors },
    setValue,
  } = useForm<FootprintFormData>({
    resolver: zodResolver(footprintFormSchema),
  })

  const { addFootprint } = useFootprint()

  const [calculateFunction, { loading }] =
    useMutation<CalculateFootprintMutationResponse>(
      CALCULATE_FOOTPRINT_MUTATION,
    )

  const handleCalculateFootprint = useCallback(
    async (formData: FootprintFormData) => {
      // handle submit
      const { data: footPrintData } = await calculateFunction({
        variables: {
          input: {
            electricityUsageKWh: Number(formData.electricty),
            transportationUsageGallonsPerMonth: Number(formData.transportation),
            flightsShortHaul: Number(formData.shortFlights),
            flightsMediumHaul: Number(formData.mediumFlights),
            flightsLongHaul: Number(formData.longFlights),
            dietaryChoice: formData.dietaryChoice,
          },
        },
      })

      addFootprint(footPrintData?.calculate.footprint as Footprint)
    },
    [calculateFunction, addFootprint],
  )

  return (
    <Card>
      <CardContent>
        <Typography variant="h5">Calculate your carbon footprint</Typography>
        <form
          onSubmit={handleSubmit(handleCalculateFootprint)}
          className="mt-4 flex w-full flex-col gap-2"
        >
          <Stack spacing={2}>
            <Input
              label="Electricity Usage (Kwh/Month):"
              type="number"
              helperText={errors.electricty?.message}
              color="secondary"
              {...register('electricty')}
            />
            <Input
              label="Transportation Gasoline Usage (Gallons/Month):"
              type="number"
              helperText={errors.transportation?.message}
              color="secondary"
              {...register('transportation')}
            />
            <Input
              label="Short Flights (4 hours):"
              type="number"
              helperText={errors.shortFlights?.message}
              color="secondary"
              {...register('shortFlights')}
            />
            <Input
              label="Medium Flights (6 hours):"
              type="number"
              helperText={errors.mediumFlights?.message}
              color="secondary"
              {...register('mediumFlights')}
            />
            <Input
              label="Long Flights (8 hours):"
              type="number"
              helperText={errors.longFlights?.message}
              color="secondary"
              {...register('longFlights')}
            />
            <SearchInput
              label="Dietary Choice:"
              helperText={errors.dietaryChoice?.message}
              color="secondary"
              onChangeValue={(value) => setValue('dietaryChoice', value)}
              {...register('dietaryChoice')}
            />
          </Stack>

          <Button
            variant="contained"
            color="secondary"
            className="mt-4"
            type="submit"
          >
            <strong>{loading ? 'Loading...' : 'Calculate'}</strong>
          </Button>
        </form>
      </CardContent>
    </Card>
  )
}
