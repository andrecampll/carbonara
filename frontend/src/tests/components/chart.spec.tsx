import { MockedProvider } from '@apollo/client/testing'
import { cleanup, screen } from '@testing-library/react'
import { afterEach, describe, expect, it } from 'vitest'

import { Chart } from '@/app/(pages)/(home)/components/chart'
import {
  FootprintContextDefaultValues,
  FootprintContextType,
} from '@/hooks/use-footprint'
import {
  footprint,
  footprintChartData,
} from '@/utils/tests/mocks/footprint.mock'
import { render } from '@/utils/tests/render'

afterEach(cleanup)

describe('Chart', () => {
  it('should be able to render Chart component', () => {
    const footprintProviderProps: FootprintContextType = {
      ...FootprintContextDefaultValues,
      footprint,
      footprintChartData,
    }

    const { container } = render(
      <MockedProvider addTypename={false}>
        <Chart />
      </MockedProvider>,
      {
        footprintProviderProps,
      },
    )

    expect(
      screen.getByRole('heading', {
        name: /Your carbon footprint/i,
      }),
    ).toBeDefined()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should be able to render Chart without unit', () => {
    const footprintProviderProps: FootprintContextType = {
      ...FootprintContextDefaultValues,
      footprint: {
        ...footprint,
        totalYearlyEmissions: {
          value: 0,
          unit: null,
        },
      },
      footprintChartData,
    }

    const { container } = render(
      <MockedProvider addTypename={false}>
        <Chart />
      </MockedProvider>,
      {
        footprintProviderProps,
      },
    )

    expect(
      screen.getByRole('heading', {
        name: /Your carbon footprint/i,
      }),
    ).toBeDefined()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should not be able to render Chart component without footprint', () => {
    const { container } = render(
      <MockedProvider addTypename={false}>
        <Chart />
      </MockedProvider>,
    )

    expect(
      screen.queryByRole('heading', {
        name: /Your carbon footprint/i,
      }),
    ).toBeNull()
    expect(container.firstChild).toMatchSnapshot()
  })
})
