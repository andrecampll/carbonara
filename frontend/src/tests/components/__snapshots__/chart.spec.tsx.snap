// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Chart > should be able to render Chart component 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
    >
      Your carbon footprint
    </h5>
    <div
      class="mt-8 flex w-full items-center"
    >
      <div
        class="css-1tfmlru-MuiResponsiveChart-container"
      >
        <svg
          class="css-13aj3tc-MuiChartsSurface-root"
          height="300"
          viewBox="0 0 600 300"
          width="600"
        >
          <title />
          <desc />
          <defs />
          <g>
            <g
              transform="translate(252.5, 150)"
            >
              <g>
                <path
                  class="MuiPieArc-root MuiPieArc-series-auto-generated-id-0 css-1mhcdve-MuiPieArc-root"
                  cursor="unset"
                  d="M27.745,-66.981L0,0Z"
                  style="fill: #007EA7; opacity: 1;"
                />
                <path
                  class="MuiPieArc-root MuiPieArc-series-auto-generated-id-0 css-1mhcdve-MuiPieArc-root"
                  cursor="unset"
                  d="M62.787,36.25L0,0Z"
                  style="fill: #1DD1A0; opacity: 1;"
                />
                <path
                  class="MuiPieArc-root MuiPieArc-series-auto-generated-id-0 css-1mhcdve-MuiPieArc-root"
                  cursor="unset"
                  d="M-36.25,62.787L0,0Z"
                  style="fill: #9F7E69; opacity: 1;"
                />
                <path
                  class="MuiPieArc-root MuiPieArc-series-auto-generated-id-0 css-1mhcdve-MuiPieArc-root"
                  cursor="unset"
                  d="M-66.981,-27.745L0,0Z"
                  style="fill: #FFCAE9; opacity: 1;"
                />
              </g>
            </g>
            <g
              transform="translate(252.5, 150)"
            >
              <g>
                <text
                  class="MuiPieArcLabel-root MuiPieArcLabel-series-auto-generated-id-0 css-ixpz1i-MuiPieArcLabel-root"
                  style="opacity: 0; transform: none;"
                />
                <text
                  class="MuiPieArcLabel-root MuiPieArcLabel-series-auto-generated-id-0 css-ixpz1i-MuiPieArcLabel-root"
                  style="opacity: 0; transform: none;"
                />
                <text
                  class="MuiPieArcLabel-root MuiPieArcLabel-series-auto-generated-id-0 css-ixpz1i-MuiPieArcLabel-root"
                  style="opacity: 0; transform: none;"
                />
                <text
                  class="MuiPieArcLabel-root MuiPieArcLabel-series-auto-generated-id-0 css-ixpz1i-MuiPieArcLabel-root"
                  style="opacity: 0; transform: none;"
                />
              </g>
            </g>
          </g>
          <g
            class="MuiChartsLegend-root MuiChartsLegend-column css-1u0lry5-MuiChartsLegend-root"
          >
            <g
              class="MuiChartsLegend-series"
              transform="translate(555 105)"
            >
              <rect
                class="MuiChartsLegend-mark"
                fill="#007EA7"
                height="20"
                width="20"
                x="0"
                y="-10"
              />
              <text
                dominant-baseline="central"
                style="font-family: "Roboto", "Helvetica", "Arial", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1; letter-spacing: 0.00938em; fill: rgba(0, 0, 0, 0.87);"
                text-anchor="start"
                x="25"
                y="0"
              >
                <tspan
                  dominant-baseline="central"
                  dy="0px"
                  x="25"
                >
                  Electricity
                </tspan>
              </text>
            </g>
            <g
              class="MuiChartsLegend-series"
              transform="translate(555 135)"
            >
              <rect
                class="MuiChartsLegend-mark"
                fill="#1DD1A0"
                height="20"
                width="20"
                x="0"
                y="-10"
              />
              <text
                dominant-baseline="central"
                style="font-family: "Roboto", "Helvetica", "Arial", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1; letter-spacing: 0.00938em; fill: rgba(0, 0, 0, 0.87);"
                text-anchor="start"
                x="25"
                y="0"
              >
                <tspan
                  dominant-baseline="central"
                  dy="0px"
                  x="25"
                >
                  Transportation
                </tspan>
              </text>
            </g>
            <g
              class="MuiChartsLegend-series"
              transform="translate(555 165)"
            >
              <rect
                class="MuiChartsLegend-mark"
                fill="#9F7E69"
                height="20"
                width="20"
                x="0"
                y="-10"
              />
              <text
                dominant-baseline="central"
                style="font-family: "Roboto", "Helvetica", "Arial", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1; letter-spacing: 0.00938em; fill: rgba(0, 0, 0, 0.87);"
                text-anchor="start"
                x="25"
                y="0"
              >
                <tspan
                  dominant-baseline="central"
                  dy="0px"
                  x="25"
                >
                  Air Travel
                </tspan>
              </text>
            </g>
            <g
              class="MuiChartsLegend-series"
              transform="translate(555 195)"
            >
              <rect
                class="MuiChartsLegend-mark"
                fill="#FFCAE9"
                height="20"
                width="20"
                x="0"
                y="-10"
              />
              <text
                dominant-baseline="central"
                style="font-family: "Roboto", "Helvetica", "Arial", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1; letter-spacing: 0.00938em; fill: rgba(0, 0, 0, 0.87);"
                text-anchor="start"
                x="25"
                y="0"
              >
                <tspan
                  dominant-baseline="central"
                  dy="0px"
                  x="25"
                >
                  Dietary Choice
                </tspan>
              </text>
            </g>
          </g>
        </svg>
      </div>
    </div>
  </div>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      Carbon Footprint Breakdown
    </h6>
    <ul
      class="mt-4"
    >
      <li
        class="flex justify-between"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1qqynwt-MuiTypography-root"
        >
          Electricity
        </p>
        <span
          class="flex items-center gap-2"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gvvk8k-MuiTypography-root"
          >
            300.00
          </p>
          <strong>
            kgCO2e/year
          </strong>
        </span>
      </li>
      <li
        class="flex justify-between"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1qqynwt-MuiTypography-root"
        >
          Transportation
        </p>
        <span
          class="flex items-center gap-2"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gvvk8k-MuiTypography-root"
          >
            1000.00
          </p>
          <strong>
            kgCO2e/year
          </strong>
        </span>
      </li>
      <li
        class="flex justify-between"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1qqynwt-MuiTypography-root"
        >
          Air Travel
        </p>
        <span
          class="flex items-center gap-2"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gvvk8k-MuiTypography-root"
          >
            200.00
          </p>
          <strong>
            kg
          </strong>
        </span>
      </li>
      <li
        class="flex justify-between"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1qqynwt-MuiTypography-root"
        >
          Dietary Choice
        </p>
        <span
          class="flex items-center gap-2"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gvvk8k-MuiTypography-root"
          >
            900.00
          </p>
          <strong>
            kgCO2e/year
          </strong>
        </span>
      </li>
      <li
        class="flex justify-between"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1qqynwt-MuiTypography-root"
        >
          Total Emissions
        </p>
        <span
          class="flex items-center gap-2"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gvvk8k-MuiTypography-root"
          >
            1000.00
          </p>
          <strong>
            kgCO2e/year
          </strong>
        </span>
      </li>
    </ul>
  </div>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Your carbon footprint is calculated based on the data you provided. The data is calculated using the latest carbon footprint calculation methods and is an estimate. The data is not 100% accurate and should be used as a reference only.
    </p>
     
    <br />
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      You can test different scenarios by changing the data you provided and see how it affects your carbon footprint. You can also see how different choices can help you reduce your carbon footprint.
    </p>
  </div>
  <div
    class="MuiCardContent-root flex w-full justify-center css-46bh2p-MuiCardContent-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-f76bo7-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Try again with different values
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`Chart > should be able to render Chart without unit 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
    >
      Your carbon footprint
    </h5>
    <div
      class="mt-8 flex w-full items-center"
    >
      <div
        class="css-1tfmlru-MuiResponsiveChart-container"
      >
        <svg
          class="css-13aj3tc-MuiChartsSurface-root"
          height="300"
          viewBox="0 0 600 300"
          width="600"
        >
          <title />
          <desc />
          <defs />
          <g>
            <g
              transform="translate(252.5, 150)"
            >
              <g>
                <path
                  class="MuiPieArc-root MuiPieArc-series-auto-generated-id-0 css-1mhcdve-MuiPieArc-root"
                  cursor="unset"
                  d="M27.745,-66.981L0,0Z"
                  style="fill: #007EA7; opacity: 1;"
                />
                <path
                  class="MuiPieArc-root MuiPieArc-series-auto-generated-id-0 css-1mhcdve-MuiPieArc-root"
                  cursor="unset"
                  d="M62.787,36.25L0,0Z"
                  style="fill: #1DD1A0; opacity: 1;"
                />
                <path
                  class="MuiPieArc-root MuiPieArc-series-auto-generated-id-0 css-1mhcdve-MuiPieArc-root"
                  cursor="unset"
                  d="M-36.25,62.787L0,0Z"
                  style="fill: #9F7E69; opacity: 1;"
                />
                <path
                  class="MuiPieArc-root MuiPieArc-series-auto-generated-id-0 css-1mhcdve-MuiPieArc-root"
                  cursor="unset"
                  d="M-66.981,-27.745L0,0Z"
                  style="fill: #FFCAE9; opacity: 1;"
                />
              </g>
            </g>
            <g
              transform="translate(252.5, 150)"
            >
              <g>
                <text
                  class="MuiPieArcLabel-root MuiPieArcLabel-series-auto-generated-id-0 css-ixpz1i-MuiPieArcLabel-root"
                  style="opacity: 0; transform: none;"
                />
                <text
                  class="MuiPieArcLabel-root MuiPieArcLabel-series-auto-generated-id-0 css-ixpz1i-MuiPieArcLabel-root"
                  style="opacity: 0; transform: none;"
                />
                <text
                  class="MuiPieArcLabel-root MuiPieArcLabel-series-auto-generated-id-0 css-ixpz1i-MuiPieArcLabel-root"
                  style="opacity: 0; transform: none;"
                />
                <text
                  class="MuiPieArcLabel-root MuiPieArcLabel-series-auto-generated-id-0 css-ixpz1i-MuiPieArcLabel-root"
                  style="opacity: 0; transform: none;"
                />
              </g>
            </g>
          </g>
          <g
            class="MuiChartsLegend-root MuiChartsLegend-column css-1u0lry5-MuiChartsLegend-root"
          >
            <g
              class="MuiChartsLegend-series"
              transform="translate(555 105)"
            >
              <rect
                class="MuiChartsLegend-mark"
                fill="#007EA7"
                height="20"
                width="20"
                x="0"
                y="-10"
              />
              <text
                dominant-baseline="central"
                style="font-family: "Roboto", "Helvetica", "Arial", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1; letter-spacing: 0.00938em; fill: rgba(0, 0, 0, 0.87);"
                text-anchor="start"
                x="25"
                y="0"
              >
                <tspan
                  dominant-baseline="central"
                  dy="0px"
                  x="25"
                >
                  Electricity
                </tspan>
              </text>
            </g>
            <g
              class="MuiChartsLegend-series"
              transform="translate(555 135)"
            >
              <rect
                class="MuiChartsLegend-mark"
                fill="#1DD1A0"
                height="20"
                width="20"
                x="0"
                y="-10"
              />
              <text
                dominant-baseline="central"
                style="font-family: "Roboto", "Helvetica", "Arial", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1; letter-spacing: 0.00938em; fill: rgba(0, 0, 0, 0.87);"
                text-anchor="start"
                x="25"
                y="0"
              >
                <tspan
                  dominant-baseline="central"
                  dy="0px"
                  x="25"
                >
                  Transportation
                </tspan>
              </text>
            </g>
            <g
              class="MuiChartsLegend-series"
              transform="translate(555 165)"
            >
              <rect
                class="MuiChartsLegend-mark"
                fill="#9F7E69"
                height="20"
                width="20"
                x="0"
                y="-10"
              />
              <text
                dominant-baseline="central"
                style="font-family: "Roboto", "Helvetica", "Arial", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1; letter-spacing: 0.00938em; fill: rgba(0, 0, 0, 0.87);"
                text-anchor="start"
                x="25"
                y="0"
              >
                <tspan
                  dominant-baseline="central"
                  dy="0px"
                  x="25"
                >
                  Air Travel
                </tspan>
              </text>
            </g>
            <g
              class="MuiChartsLegend-series"
              transform="translate(555 195)"
            >
              <rect
                class="MuiChartsLegend-mark"
                fill="#FFCAE9"
                height="20"
                width="20"
                x="0"
                y="-10"
              />
              <text
                dominant-baseline="central"
                style="font-family: "Roboto", "Helvetica", "Arial", sans-serif; font-weight: 400; font-size: 1rem; line-height: 1; letter-spacing: 0.00938em; fill: rgba(0, 0, 0, 0.87);"
                text-anchor="start"
                x="25"
                y="0"
              >
                <tspan
                  dominant-baseline="central"
                  dy="0px"
                  x="25"
                >
                  Dietary Choice
                </tspan>
              </text>
            </g>
          </g>
        </svg>
      </div>
    </div>
  </div>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      Carbon Footprint Breakdown
    </h6>
    <ul
      class="mt-4"
    >
      <li
        class="flex justify-between"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1qqynwt-MuiTypography-root"
        >
          Electricity
        </p>
        <span
          class="flex items-center gap-2"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gvvk8k-MuiTypography-root"
          >
            300.00
          </p>
          <strong>
            kgCO2e/year
          </strong>
        </span>
      </li>
      <li
        class="flex justify-between"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1qqynwt-MuiTypography-root"
        >
          Transportation
        </p>
        <span
          class="flex items-center gap-2"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gvvk8k-MuiTypography-root"
          >
            1000.00
          </p>
          <strong>
            kgCO2e/year
          </strong>
        </span>
      </li>
      <li
        class="flex justify-between"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1qqynwt-MuiTypography-root"
        >
          Air Travel
        </p>
        <span
          class="flex items-center gap-2"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gvvk8k-MuiTypography-root"
          >
            200.00
          </p>
          <strong>
            kg
          </strong>
        </span>
      </li>
      <li
        class="flex justify-between"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1qqynwt-MuiTypography-root"
        >
          Dietary Choice
        </p>
        <span
          class="flex items-center gap-2"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gvvk8k-MuiTypography-root"
          >
            900.00
          </p>
          <strong>
            kgCO2e/year
          </strong>
        </span>
      </li>
      <li
        class="flex justify-between"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1qqynwt-MuiTypography-root"
        >
          Total Emissions
        </p>
        <span
          class="flex items-center gap-2"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gvvk8k-MuiTypography-root"
          >
            0.00
          </p>
          <strong>
            kg
          </strong>
        </span>
      </li>
    </ul>
  </div>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Your carbon footprint is calculated based on the data you provided. The data is calculated using the latest carbon footprint calculation methods and is an estimate. The data is not 100% accurate and should be used as a reference only.
    </p>
     
    <br />
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      You can test different scenarios by changing the data you provided and see how it affects your carbon footprint. You can also see how different choices can help you reduce your carbon footprint.
    </p>
  </div>
  <div
    class="MuiCardContent-root flex w-full justify-center css-46bh2p-MuiCardContent-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-f76bo7-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Try again with different values
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`Chart > should not be able to render Chart component without footprint 1`] = `null`;
